src/compiler/checker.ts

```diff
@@ -9664,6 +9677,13 @@ namespace ts {
             }
         }
 
+        function getIndexInfoOfIndexSymbol(indexSymbol: Symbol, indexKind: IndexKind) {

```

Comment:
Since this is used only once in the above changed code, maybe make it a local helper instead?

Reply:
I kinda prefer it this way to avoid several closure allocations, but it's likely a premature optimization.

Reply:
(I'd be surprised if at least the v8 compiler wouldn't optimize it away...)